python/src/netlist.py

```diff
@@ -13,18 +13,25 @@
 
 
 def _extract_float(line: str, regex: str) -> float:
```

Comment:
nit: while we're refactoring, I think we might be able to do a generic extract fn

```python3
from typing import Type, TypeVar

T = TypeVar('T')

def _extract(line: str, regex: str, t: Type[T]) -> T:
    match = re.search(regex, line, re.IGNORECASE)
    if match:
        str_value = match.group(1)
        return t(str_value)
    raise ValueError(f"No match of {regex} found in {line}")
```

then usages look like
```python3
_extract(l, DEVICE_WIDTH_REGEX, float)
```